service/pager/query_pagers.cc

```diff
@@ -57,7 +58,21 @@ query_pager::query_pager(service::storage_proxy& p, schema_ptr s,
                 , _options(options)
                 , _cmd(std::move(cmd))
                 , _ranges(std::move(ranges))
-{}
+{
+    if (query_function_override) {
+        _query_function = std::move(query_function_override);
+    } else {
+        _query_function = [] (
+                service::storage_proxy& sp,
+                schema_ptr schema,
+                lw_shared_ptr<query::read_command> cmd,
+                dht::partition_range_vector&& partition_ranges,
+                db::consistency_level cl,
+                service::storage_proxy_coordinator_query_options optional_params) {
+            return sp.query_result(std::move(schema), std::move(cmd), std::move(partition_ranges), cl, std::move(optional_params));
+        };
+    }
+}
```

Comment:
Please run perf_simple_query to see this doesn't generate some silly regression.